aten/src/ATen/templates/Functions.h

```diff
@@ -20,14 +21,22 @@ using native::tensor;
 
 ${function_declarations}
 
-static inline Type & infer_type(const Tensor & t) {
+namespace detail {
+
+static inline TypeExtendedInterface & infer_type(const Tensor & t) {
   AT_CHECK(t.defined(), "undefined Tensor");
-  return t.type();
+  return static_cast<TypeExtendedInterface&>(t.type());
 }
-static inline Type & infer_type(const TensorList & tl) {
+static inline TypeExtendedInterface & infer_type(const TensorList & tl) {
   AT_CHECK(tl.size() > 0, "expected a non-empty list of Tensors");
-  return tl[0].type();
+  return static_cast<TypeExtendedInterface&>(tl[0].type());
+}
+static inline TypeExtendedInterface & non_specific_type() {
+  return static_cast<TypeExtendedInterface&>(at::getNonVariableType(at::Backend::Undefined, at::ScalarType::Float));
```

Comment:
this is just bogus (that type doesn't even exist), but I guess it existed before.